trestle/core/commands/replicate.py

```diff
@@ -14,11 +14,234 @@
+        return ReplicateCmd.replicate_object(self.name, component.ComponentDefinition, args)
+
+
+class SystemSecurityPlanCmd(Command):
+    """Replicate a system security plan within the trestle directory structure."""
+
+    name = 'system-security-plan'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, ssp.SystemSecurityPlan, args)
+
+
+class AssessmentPlanCmd(Command):
+    """Replicate an assessment plan within the trestle directory structure."""
+
+    name = 'assessment-plan'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, assessment_plan.AssessmentPlan, args)
+
+
+class AssessmentResultCmd(Command):
+    """Replicate an assessment result within the trestle directory structure."""
+
+    name = 'assessment-results'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, assessment_results.AssessmentResults, args)
+
+
+class PlanOfActionAndMilestonesCmd(Command):
+    """Replicate a plan of action and milestones within the trestle directory structure."""
+
+    name = 'plan-of-action-and-milestones'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, poam.PlanOfActionAndMilestones, args)
+
 
 class ReplicateCmd(Command):
-    """Replicate an existing trestle model generating new UUIDs."""
+    """Replicate an existing top level OSCAL model within the trestle project."""
 
     name = 'replicate'
+
+    subcommands = [
+        CatalogCmd,
+        ProfileCmd,
+        TargetDefinitionCmd,
+        ComponentDefinitionCmd,
+        SystemSecurityPlanCmd,
+        AssessmentPlanCmd,
+        AssessmentResultCmd,
+        PlanOfActionAndMilestonesCmd
+    ]
+
+    def _init_arguments(self) -> None:
+        logger.debug('Init arguments')
+        self.add_argument('-f', '--file', help='OSCAL model to replicate.', type=str, required=True)
+
+        self.add_argument('-o', '--output', help='Name of replicated model.', type=str, required=True)
+
+        self.add_argument(
+            '-r', '--regenerate', type=bool, default=False, help='Enable to regenerate uuids within the document'
+        )
+
+    @classmethod
+    def replicate_object(cls, model_alias: str, object_type: Type[TLO], args: argparse.Namespace) -> int:
+        """Core replicate routine invoked by subcommands."""
+        log.set_log_level_from_args(args)
+
+        logger.debug('Entering replicate_object.')
+
+        # 1.2 Bad working directory if not running from current working directory
+        cwd = pathlib.Path.cwd().resolve()
+        trestle_root = fs.get_trestle_project_root(cwd)
+        if trestle_root is None:
+            logger.error(f'Current working directory: {cwd} is not within a trestle project.')
+            return 1
+
+        trestle_root = trestle_root.resolve()
+
+        plural_path: str
```

Comment:
this definition is redundant. You typically only need definitons in scenarios where you have to declare where the type is not obvious..
e.g 
`array_strings: List[str] = []`
as the type can no be inferred.